monai/bundle/scripts.py

```diff
@@ -932,6 +939,93 @@ def _export(
     logger.info(f"exported to file: {filepath}.")
 
 
+def onnx_export(
+    net_id: str | None = None,
+    filepath: PathLike | None = None,
+    ckpt_file: str | None = None,
+    meta_file: str | Sequence[str] | None = None,
+    config_file: str | Sequence[str] | None = None,
+    key_in_ckpt: str | None = None,
+    use_trace: bool | None = None,
+    input_shape: Sequence[int] | None = None,
+    args_file: str | None = None,
+    **override: Any,
+) -> None:
+    """
+    Export the model checkpoint to onnx with metadata and config included as JSON files.
+
+    Typical usage examples:
+
+    .. code-block:: bash
+
+        python -m monai.bundle onnx_export network --filepath <export path> --ckpt_file <checkpoint path> ...
+
+    Args:
+        net_id: ID name of the network component in the config, it must be `torch.nn.Module`.
+        filepath: filepath where the onnx model is saved to.
+        ckpt_file: filepath of the model checkpoint to load.
+        meta_file: filepath of the metadata file, if it is a list of file paths, the content of them will be merged.
+        config_file: filepath of the config file that contains extract network information,
+        key_in_ckpt: for nested checkpoint like `{"model": XXX, "optimizer": XXX, ...}`, specify the key of model
+            weights. if not nested checkpoint, no need to set.
+        use_trace: whether using `torch.jit.trace` to convert the pytorch model to torchscript model.
+        input_shape: must specify the `input_shape` of the network to convert the model when `use_trace` is True.
+            Should be a list like [N, C, H, W] or [N, C, H, W, D].
+        args_file: a JSON or YAML file to provide default values for `meta_file`, `config_file`,
+            `net_id` and override pairs. so that the command line inputs can be simplified.
+        override: id-value pairs to override or add the corresponding config content.
+            e.g. ``--_meta#network_data_format#inputs#image#num_channels 3``.
+
+    """
+    onnx, _ = optional_import("onnx")
+
+    _args = _update_args(
+        args=args_file,
+        net_id=net_id,
+        filepath=filepath,
+        meta_file=meta_file,
+        config_file=config_file,
+        ckpt_file=ckpt_file,
+        key_in_ckpt=key_in_ckpt,
+        use_trace=use_trace,
+        input_shape=input_shape,
+        **override,
+    )
+    _log_input_summary(tag="onnx_export", args=_args)
+    filepath_, ckpt_file_, config_file_, net_id_, meta_file_, _, _, input_shape_ = _pop_args(
+        _args,
+        "filepath",
+        "ckpt_file",
+        "config_file",
+        net_id="",
+        meta_file=None,
+        key_in_ckpt="",
+        use_trace=True,
```

Comment:
the default should be `False`?

Thanks.

Reply:
PyTorch trace is in general more reliable than scripting. scripting is useful for loop and if which is no comment in computer vision. But if it is a convention in MONAI, we can change default to False.

Reply:
I would suggest to use `jit.script()` if possible, especially for dynamic input shapes.
@ericspod Do you have any comments?

Thanks.

Reply:
We should use `jit.script` where we can but there's times that we would want to use `jit.trace`. Not all networks are compatible with `jit.script` (or at least weren't in the past with previous versions of Pytorch) so we need that option, I'm not aware of any other reason to use one function or the other but they must exist. Networks created with `jit.script` appear compatible with ONNX from my testing.

Reply:
Yes, that's why I would suggest to set the default to `False`, and I agree we need this parameter.

Thanks.